keydata.go

```diff
@@ -52,6 +55,92 @@ const (
 	AuthModePIN
 )
 
+type afSplitDataRawHdr struct {
+	Stripes uint32
+	HashAlg tpm2.HashAlgorithmId
+	Size    uint32
+}
+
+type afSplitDataRaw struct {
+	Hdr  afSplitDataRawHdr
+	Data tpm2.RawBytes
+}
+
+func (d *afSplitDataRaw) Marshal(w io.Writer) (nbytes int, err error) {
+	return tpm2.MarshalToWriter(w, d.Hdr, d.Data)
+}
+
+func (d *afSplitDataRaw) Unmarshal(r io.Reader) (nbytes int, err error) {
+	var h afSplitDataRawHdr
+	n, err := tpm2.UnmarshalFromReader(r, &h)
+	nbytes += n
+	if err != nil {
+		return nbytes, xerrors.Errorf("cannot unmarshal header: %w", err)
+	}
+
+	data := make([]byte, h.Size)
+	n, err = io.ReadFull(r, data)
+	nbytes += n
+	if err != nil {
+		return nbytes, xerrors.Errorf("cannot read data: %w", err)
```

Comment:
What happens to the memory used by `data` in the event this fails?

Reply:
It should eventually be reclaimed by go's garbage collector.